GMG/Scenes/ChordProgress/Waveform.swift

```diff
@@ -0,0 +1,181 @@
+            self.maxCapsuleHeight = maxCapsuleHeight
+        }
+        
+        var body: some View {
+            HStack(alignment: .center, spacing: capsuleSpacing) {
+                ForEach(amplitudes.indices, id: \.self) { index in
+                    let amplitude = amplitudes[index]
+                    Capsule()
+                        .fill(color)
+                        .frame(
+                            width: capsuleWidth,
+                            height: capsuleHeight(for: amplitude)
+                        )
+                }
+            }
+            .frame(maxWidth: .infinity, maxHeight: .infinity)
+            .padding(.horizontal, horizontalPadding)
+
+        }
+        
+        func capsuleHeight(for amplitude: Float) -> CGFloat {
+            let clamped = max(0, min(1, amplitude))
+            let range = maxCapsuleHeight - minCapsuleHeight
+            return minCapsuleHeight + range * CGFloat(clamped)
+        }
+    }
+    
+    private func fillWidth(totalWidth: CGFloat) -> CGFloat {
+        guard endTime > startTime else { return 0 }
+        let clampedElapsed = min(max(elapsedTime, startTime), endTime)
+        let ratio = (clampedElapsed - startTime) / (endTime - startTime)
+        return totalWidth * CGFloat(ratio)
+    }
+    
+    private func capsuleCount(for totalWidth: CGFloat) -> Int {
+        let contentWidth = max(0, totalWidth - (horizontalPadding * 2))
+        let unit = capsuleWidth + capsuleSpacing
+        guard unit > 0 else { return max(1, amplitudes.count) }
+        
+        let count = Int(floor((contentWidth + capsuleSpacing) / unit))
+        return max(1, count)
+    }
+    
+    private func amplitudesForRendering(targetCount: Int) -> [Float] {
+        guard targetCount > 0 else { return [] }
+        
+        if amplitudes.isEmpty {
+            return Array(repeating: 0.0, count: targetCount)
+        }
+        
+        if amplitudes.count == targetCount {
+            return amplitudes
+        } else if amplitudes.count > targetCount {
+            return resample(amplitudes, to: targetCount)
+        } else {
+            return resample(amplitudes, to: targetCount)
+        }
+    }
+    
+    private func resample(_ source: [Float], to targetCount: Int) -> [Float] {
+        guard targetCount > 1, source.count > 1 else {
+            let value = source.first ?? 0
+            return Array(repeating: value, count: targetCount)
+        }
+        
+        return (0..<targetCount).map { index in
+            let position = Float(index) / Float(targetCount - 1)
+            let scaled = position * Float(source.count - 1)
+            let lower = Int(floor(scaled))
+            let upper = Int(ceil(scaled))
+            
+            if lower == upper {
+                return source[lower]
+            }
+            
+            let t = scaled - Float(lower)
+            return source[lower] * (1 - t) + source[upper] * t
```

Comment:
`t`의 의미를 명확하게 알 수 있도록 이름 바꾸는 것은 어떨까요?

```suggestion
            let interpolationFactor = scaled - Float(lower)
            return source[lower] * (1 - interpolationFactor) + source[upper] * interpolationFactor
```

Reply:
좋습니다~!